### DISCUSSION TITLE
'Unable to fini publisher for node., at ./src/rcl/node.c:380'

### DISCUSSION THREAD

**Discussion by wtokumaru** [eProsima/Fast-DDS#5222] [Support]
Does anyone know where the discussion in this ticket was moved to by @JLBuenoLopez ?: https://github.com/eProsima/Fast-DDS/discussions/3737
I have not found it via search since the only result is the same original discussion.
I am getting a very similar issue and still get it when `is_default_profile` set to `"true"`. 
My specific use-case is potentially simpler as I am using the example here to try to set any xml profile: https://fast-dds.docs.eprosima.com/en/latest/fastdds/ros2/ros2_configure.html

**Comment by JesusPoderoso** [comment]
Hi @wtokumaru, the thread was originally an "issue" ticket, in the [Issues](https://github.com/eProsima/Fast-DDS/issues) section, and was moved to the [Discussions](https://github.com/eProsima/Fast-DDS/discussions) section, in particular to the [support section](https://github.com/eProsima/Fast-DDS/discussions/categories/support). So the thread you refer to is the original and complete thread but hosted in different GitHub sections.

Are you following the XML profiles path and environment variables as explained in [this section](https://fast-dds.docs.eprosima.com/en/latest/fastdds/ros2/ros2_configure.html#xml-configuration-file-location)?
Which ROS 2 version (and Fast DDS version) are you using? Is there any reproducer that you could provide to us for replicating your issue?

**Reply by wtokumaru** [reply]
Hello Jesus, thank you for reaching out! It turns out I am using ROS humble and was testing with DDS 3.0 so when I swapped back to 2.6, the code was able to run. I was also able to get similar results with this: https://github.com/eProsima/rmw_fastrtps/tree/humble?tab=readme-ov-file#example

However, I am unable to set any of the helloworld channel QoS differently from the default. Could you perhaps share an example of this file which causes the QoS to change? I have been referencing [11.8. Common — Fast DDS 2.6.9 documentation (eprosima.com)](https://fast-dds.docs.eprosima.com/en/v2.6.9/fastdds/xml_configuration/common.html#commonqos)

For example, if I insert the following into the helloworld <qos> section, the reliability still reports as BEST_EFFORT when I ros2 topic info -v /helloworld:

```
    <reliability>
        <kind>RELIABLE</kind>
    </reliability>

```

This goes the same for all of these settings, so I suspect I am missing some other step beyond the envvars. I know that the file is being read and used because if I set anything invalid, it crashes.

Additionally, I get many warning messages which tend to look like this when I use the profile, even with default QoS:

```
2024-09-11 16:44:49.489 [RTPS_READER_HISTORY Error] Change payload size of '56' bytes is larger than the history payload size of '35' bytes and cannot be resized. -> Function can_change_be_added_nts

```
But then otherwise it runs normally.

**Comment by Ryanf55** [comment]
Is there any way that eProsima could add a small demonstration on the existing tutorial for [ros2-configure](https://fast-dds.docs.eprosima.com/en/latest/fastdds/ros2/ros2_configure.html) on how to set up a topic like the `Static Transform Broadcaster` that uses transient local durability with history length 1, another topic with "Best effort" reliability, and finally a "reliable" publisher?

In essense, I want a simple demonstration that shows how to take the [standard QOS's used in ROS2](https://docs.ros.org/en/rolling/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-profiles) and replicate them in XML.
This is the first step for anyone looking to migrate off of the ROS 2 API's for QOS into FastDDS XML configuration.
